coding-guides/using-audit-logs.md

```diff
@@ -0,0 +1,142 @@
+})
+```
+Lets take a break to explain exactly whats going on. The executor will be retrieving the very latest audit log entry and all of its information that goes with it. What does this information contain? Everything we need for logging. 
+
+```js
+GuildAuditLogsEntry {
+  targetType: 'MESSAGE',
+  actionType: 'DELETE',
+  action: 'MESSAGE_DELETE',
+  reason: null,
+  executor: [Object],
+  changes: null,
+  id: '434116792567201792',
+  extra: [Object],
+  target: [Object] }
+```
+Notice the `reason` field. Some audit logs, like kicking and banning, can provide a reason. You can probably make logs of when a user is banned and for whatever reason.
+What we want is the executor of the action. We do that by going to the `executor` target as that is where the user is stored.
+So, our `executor` should be look similar to this:
+
+```js
+  const executor = await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(audit => audit.entries.first().executor.username)
+```
+
+Wait wait! Where did the `.username` come from?
+Excellent question! When you get the executor's information, it is considered a user.
+We can get information about that executor, like their id, avatar, and more. 
+
+```js
+User {
+  id: '286270602820452353',
+  username: 'Zoth The Wumpus',
+  discriminator: '6066',
+  avatar: '57361ef0f8e23e02a44069c3dd5f5f47',
+  bot: false,
+  lastMessageID: '435165896198062091',
+  lastMessage: [Object] }
+```
+
+So now that we have the following:
+```js
+// fetching audit logs is a promise, so we want to async the event
+client.on('messageDeleted', async (message) => {
+  // Firstly, we need a logs channel. 
+  const logs = message.guild.channels.find('name', 'logs');
+  
+  // If there is no logs channel, we can create it if we have the 'MANAGE_CHANNELS' permission
+  if (message.guild.me.hasPermission('MANAGE_CHANNELS') && !logs) {
+    message.guild.createChannel('logs', 'text');
+  }
+  
+  // If we do not have permissions, console log both errors
+  if (!message.guild.me.hasPermission('MANAGE_CHANNELS') && !logs) { 
+    console.log('The logs channel does not exist and tried to create the channel but am lacking permissions')
+  }
+  
+  // Now that we have the logs channel created...hopefully...we can send messages to it
+  // before we do that, lets get started to viewing the audit logs. 
+  // We want the user who deleted the message
+  // The "type" is how you will be searching through the audit logs. Like roles updated or members banned.
+  const executor = await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(audit => audit.entries.first().executor.username)
+})
+```
+lets start sending it all to a channel.
+
+```js
+  // I always format my messages in strings as it's easier (for me) to understand what I am doing as some strings can get pretty long,
+  // like the $5 foot long from subway. Man I miss that. 
+  const deletedMessageInformation = "A message was deleted in " + message.channel.name + " by " + executor;
```

Comment:
please learn template literals, and use them. We don't do string concatenation in this guide.